# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/DocAccessibleChild.cpp
# Commit: 2010ac6007c8
# Full Hash: 2010ac6007c8ef7684a5fbef2903fc42aa3e241d
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-03-25 03:42:26
# Description:
#   Bug 1691930 - Check that FocusedChild is bound to a doc in RecvFocusedChild. r=Jamie
#   
#   Added an assert in focus manager. Hopefully fuzzers will help us find
#   cases where the active item is defunct, if that is indeed what is
#   happening.
# ==============================================================================

diff -r 882197c37c04 -r 2010ac6007c8 accessible/ipc/other/DocAccessibleChild.cpp
--- a/accessible/ipc/other/DocAccessibleChild.cpp	Wed Mar 24 18:10:34 2021 +0200
+++ b/accessible/ipc/other/DocAccessibleChild.cpp	Wed Mar 24 16:24:47 2021 +0000
@@ -1542,6 +1542,11 @@
     // document, not just a descendant of the caller's document. Check that it
     // is really a descendant.
     DocAccessible* doc = result->Document();
+    if (!doc) {
+      MOZ_ASSERT_UNREACHABLE("Focused child is unbound from doc.");
+      return IPC_OK();
+    }
+
     while (doc != mDoc) {
       doc = doc->ParentDocument();
       if (!doc) {
